.github/workflows/compound-java-dev.yml

```diff
@@ -88,11 +105,31 @@ jobs:
     with:
       runsOn: ${{ inputs.runsOn }}
 
+  java-check:
+    name: 'Java check'
+    needs: [prebuild-job]
+    uses: th2-net/.github/.github/workflows/compound-java-check.yml@integration-tests # FIXME: switch to main branch
+    with:
+      runs-on: ${{ inputs.runsOn }}
+      app-version: ${{ needs.prebuild-job.outputs.version }}
+      java-version: ${{ inputs.javaVersion }}
+      java-vendor: ${{ inputs.javaVendor }}
+      scanner-enabled: ${{ inputs.scanner-enabled }}
+      strict-scanner: ${{ inputs.strict-scanner }}
+      integration-test-enabled: ${{ inputs.integration-test-enabled }}
+      integration-test-projects: ${{ inputs.integration-test-projects }}
+      integration-test-task: ${{ inputs.integration-test-task }}
+    secrets:
+      nvd-api-key: ${{ secrets.nvd-api-key }}
+      cisa-domain: ${{ secrets.cisa-domain }}
+      cisa-user: ${{ secrets.cisa-user }}
+      cisa-password: ${{ secrets.cisa-password }}
+
   # if Docker
   docker-push:
     name: Build Docker image and push to ghcr.io
-    if: (contains(inputs.build-target, 'Docker') || contains(inputs.build-target, 'docker')) && (inputs.strict-scanner && success() || !inputs.strict-scanner)
-    needs: [prebuild-job, owasp-scan, license-check]
+    if: ${{ (contains(inputs.build-target, 'Docker') || contains(inputs.build-target, 'docker')) && !cancelled() && needs.java-check.outputs.aggregated-status == 'true' }}
```

Comment:
note: you don't need to add `${{ ... }}` for `if`. It is explicitly treated as an expression, and the expression does not start with `!`